From 41811f5e37965ddeb6787a50c4b3a8ac09ee7a70 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=AF=9B=E4=BA=9A=E7=90=9B?= <maoyachen55@gmail.com>
Date: Wed, 16 Apr 2025 18:13:15 +0800
Subject: [PATCH 1/2] fix #352

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9b2935a6..43a90e01 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -183,7 +183,9 @@ if(BUILD_PYTHON_WRAPPER AND Python3_Development_FOUND AND Python3_NumPy_FOUND)
     set(apriltag_py_target "apriltag.${Python3_SOABI}")
     Python3_add_library(${apriltag_py_target} MODULE ${CMAKE_CURRENT_SOURCE_DIR}/apriltag_pywrap.c)
     add_dependencies(${apriltag_py_target} apriltag_py_docstrings)
-    target_link_libraries(${apriltag_py_target} PRIVATE apriltag Python3::Python Python3::NumPy)
+    # avoid linking against Python3::Python to prevent segmentation faults in Conda environments
+    # https://github.com/AprilRobotics/apriltag/issues/352
+    target_link_libraries(${apriltag_py_target} PRIVATE apriltag Python3::NumPy)
     target_include_directories(${apriltag_py_target} PRIVATE ${PROJECT_BINARY_DIR})
 
     set(PY_DEST ${CMAKE_INSTALL_PREFIX}/lib/python${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}/site-packages/)

From 018a94a9736507c1a7800e4d17d32abd15cc7593 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E6=AF=9B=E4=BA=9A=E7=90=9B?= <maoyachen55@gmail.com>
Date: Thu, 17 Apr 2025 21:43:08 +0800
Subject: [PATCH 2/2] set linker flags only for target apriltag

---
 CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 43a90e01..4806fdea 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,10 +58,6 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
     add_compile_options(-Wno-shift-negative-value)
 endif()
 
-if(CMAKE_C_COMPILER_ID MATCHES "Clang" AND NOT APPLE AND NOT CMAKE_C_SIMULATE_ID MATCHES "MSVC")
-    add_link_options("-Wl,-z,relro,-z,now,-z,defs")
-endif()
-
 if(CMAKE_C_COMPILER_ID MATCHES "Clang" AND CMAKE_C_SIMULATE_ID MATCHES "MSVC")
     # error: 'strdup' is deprecated: The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _strdup.
     # "strdup" is standard since C23
@@ -78,6 +74,10 @@ file(GLOB TAG_FILES ${CMAKE_CURRENT_SOURCE_DIR}/tag*.c)
 add_library(${PROJECT_NAME} ${APRILTAG_SRCS} ${COMMON_SRC} ${TAG_FILES})
 set_property(TARGET ${PROJECT_NAME} PROPERTY POSITION_INDEPENDENT_CODE ON)
 
+if(CMAKE_C_COMPILER_ID MATCHES "Clang" AND NOT APPLE AND NOT CMAKE_C_SIMULATE_ID MATCHES "MSVC")
+    target_link_options(${PROJECT_NAME} PRIVATE "-Wl,-z,relro,-z,now,-z,defs")
+endif()
+
 if (MSVC)
     add_compile_definitions("_CRT_SECURE_NO_WARNINGS")
 else()
