diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b238324..e8439c01 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -186,7 +186,12 @@ if(BUILD_PYTHON_WRAPPER AND Python3_Development_FOUND AND Python3_NumPy_FOUND)
     )
 
     # set the SOABI manually since renaming the library via OUTPUT_NAME does not work on MSVC
-    set(apriltag_py_target "apriltag.${Python3_SOABI}")
+    set(_apriltag_soabi "${Python3_SOABI}")
+    if(NOT _apriltag_soabi AND APRILTAG_SOABI)
+        set(_apriltag_soabi "${APRILTAG_SOABI}")
+    endif()
+
+    set(apriltag_py_target "apriltag.${_apriltag_soabi}")
     Python3_add_library(${apriltag_py_target} MODULE ${CMAKE_CURRENT_SOURCE_DIR}/apriltag_pywrap.c)
     add_dependencies(${apriltag_py_target} apriltag_py_docstrings)
     # avoid linking against Python3::Python to prevent segmentation faults in Conda environments
